### ISSUE TITLE
[Bug] HBM usage is unbalanced when loading the model

### ISSUE THREAD

**Issue by pengchengneo** [sgl-project/sglang-jax#151]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [ ] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [ ] 5. Please use English, otherwise it will be closed.

### Describe the bug

HBM usage is unbalanced when loading the model,  it will raise 'OOM' exception when load large model

### Reproduction

JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen3-8B --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=4 --device=tpu --random-seed=3  --mem-fraction-static=0.7 --chunked-prefill-size=2048 --download-dir=/tmp --dtype=bfloat16 --max-running-requests 256 --skip-server-warmup --page-size=128  --disable-radix-cache --node-rank 0

HBM usage is unbalanced when start load model, 

<img width="3840" height="2160" alt="Image" src="https://github.com/user-attachments/assets/e2adc53a-ceeb-44db-9b7b-9b56b1645fef" />

### Environment

`Package                 Version     Editable project location
----------------------- ----------- -------------------------------
absl-py                 2.3.1
aiofiles                24.1.0
aiohappyeyeballs        2.6.1
aiohttp                 3.12.15
aiosignal               1.4.0
annotated-types         0.7.0
anyio                   4.10.0
archspec                0.2.1
async-timeout           5.0.1
attrs                   25.3.0
boltons                 23.0.0
Brotli                  1.0.9
certifi                 2023.11.17
cffi                    1.16.0
charset-normalizer      2.0.4
chex                    0.1.90
click                   8.2.1
conda                   23.11.0
conda-content-trust     0.2.0
conda-libmamba-solver   23.12.0
conda-package-handling  2.2.0
conda_package_streaming 0.9.0
cryptography            41.0.7
distro                  1.8.0
etils                   1.13.0
exceptiongroup          1.3.0
fastapi                 0.116.1
filelock                3.19.1
flax                    0.10.7
frozenlist              1.7.0
fsspec                  2025.9.0
grpcio                  1.74.0
h11                     0.16.0
hf-xet                  1.1.9
httpcore                1.0.9
httpx                   0.28.1
huggingface-hub         0.34.4
humanize                4.13.0
idna                    3.4
importlib_resources     6.5.2
jax                     0.6.2
jaxlib                  0.6.2
Jinja2                  3.1.6
jiter                   0.10.0
jsonpatch               1.32
jsonpointer             2.1
libmambapy              1.5.3
libtpu                  0.0.17
markdown-it-py          4.0.0
MarkupSafe              3.0.2
mdurl                   0.1.2
menuinst                2.0.1
ml_dtypes               0.5.3
modelscope              1.28.2
msgpack                 1.1.1
msgpack-python          0.5.6
multidict               6.6.4
nest-asyncio            1.6.0
numpy                   2.2.6
openai                  1.107.0
opt_einsum              3.4.0
optax                   0.2.5
orbax-checkpoint        0.11.24
orjson                  3.11.3
packaging               23.1
pandas                  2.3.2
pillow                  11.3.0
pip                     23.3.1
platformdirs            3.10.0
pluggy                  1.0.0
propcache               0.3.2
protobuf                6.32.0
psutil                  7.0.0
pybase64                1.4.2
pycosat                 0.6.6
pycparser               2.21
pydantic                2.11.7
pydantic_core           2.33.2
Pygments                2.19.2
pyOpenSSL               23.2.0
PySocks                 1.7.1
python-dateutil         2.9.0.post0
python-multipart        0.0.20
pytz                    2025.2
PyYAML                  6.0.2
pyzmq                   27.0.2
regex                   2025.9.1
requests                2.32.5
rich                    14.1.0
ruamel.yaml             0.17.21
ruamel.yaml.clib        0.2.6
safetensors             0.6.2
scipy                   1.15.3
setproctitle            1.3.7
setuptools              68.2.2
sglang-jax              0.0.1       /home/gcpuser/sglang-jax/python
simplejson              3.20.1
six                     1.17.0
sniffio                 1.3.1
starlette               0.47.3
tensorstore             0.1.76
tiktoken                0.11.0
tokenizers              0.21.4
toolz                   1.0.0
tpu-info                0.5.1
tqdm                    4.65.0
transformers            4.55.4
treescope               0.1.10
truststore              0.8.0
typing_extensions       4.14.1
typing-inspection       0.4.1
tzdata                  2025.2
urllib3                 1.26.18
uvicorn                 0.35.0
uvloop                  0.21.0
wheel                   0.41.2
yarl                    1.20.1
zipp                    3.23.0
zstandard               0.19.0`
